dependencies/common/install-boost

```diff
@@ -26,7 +26,7 @@ PLATFORM=`uname`
 # vars
 BOOST_VERSION_NUMBER=1.69.0
 BOOST_VERSION=boost_1_69_0
-BOOST_DIR=/opt/rstudio-tools/boost/$BOOST_VERSION
+BOOST_DIR=/opt/rstudio-tools/$(uname -m)/boost/$BOOST_VERSION
```

Comment:
Note that this change would require rebuilding dependencies on x86_64 machines. Do you think that's fine, or should we consider setting this only for arm64?

Reply:
I think it's fine. It does imply that we will need to rebuild deps manually on the build host, right?